web/src/settings_exports.ts

```diff
@@ -248,3 +409,67 @@ export function set_up(): void {
         });
     });
 }
+
+function maybe_store_export_consent_data_and_return(export_consent: ExportConsent): boolean {
+    // Handles a race where the client has requested the server for export_consents
+    // to populate 'Export permissions' table but hasn't received the response yet,
+    // but received a few updated events which should be applied on top of the received
+    // response to avoid outdated table.
+    // We store the export_consent data received via events to apply them on top of
+    // the received response.
+    if (export_consents === undefined) {
+        queued_export_consents.push(export_consent);
```

Comment:
`queued_export_consents` would also be a lot cleaner as a `Map`, since then you can just walk this map, overwriting onto the `Map` that we got from the API response.

Reply:
> `queued_export_consents` would also be a lot cleaner as a Map

Kept it as an array because it's element is of type `ExportConsent | number`. We only need the key `user_id` when we want to remove the element.

We simply walk the array and perform the update or delete operation on the map. -- O(n)